articles/integrate-tailwindcss-into-flask/index.md

```diff
@@ -0,0 +1,232 @@
+
+# Integrating TailwindCSS into Flask Apps
+
+## Introduction
+
+TailwindCss is a utility-first CSS framework for building frontend applications, TailwindCss differs from other kinds of CSS frameworks, as it gives the user total control over their design, rather than adding obscure CSS classes to your code, with TailwindCss you use utility classes to create your components, with as much control over every single styling as you want, all without having to ever write a single line of CSS. 
+
+### Pre-Requisites
+For the purposes of demonstration in this article, I'm using python3, flask and JavaScript package managers (npm or yarn). I am assuming the reader has basic understanding of python and by extension - Flask, and is already quite conversant with TailwindCSS.
+
+* Python3
+* Node.js 12.13.0 or higher
+* npm or yarn
+* Flask
+* Your Favorite text editor
+* Knowledge of using TailwindCSS
+
+## Building A Simple Flask App
+
+### Installing Flask
+
+We'll begin by creating a really simple flask server, that is capable of rendering frontend templates on our behalf. To begin we open up the terminal and install flask using python's package manager, as follows:
+
+```bash
+pip install flask
+```
+
+This will install flask neatly for us on our computer, if you already have it installed you can skip this step. You can also choose to install it in a virtual environment if you feel like it.
+
+### Creating the Flask Server
+
+The next step is to create a simple flask server that will render the frontend templates for us. Create a new folder (anywhere you wish) to hold our files and create a new file called `app.py` in the folder, as well as a sub-folder called `templates`; this is where all HTML files will go, lastly we'll create another subfolder called `static`, this folder will hold all static files, including CSS files, scripts and images for our frontend templates. When you're done the folder should look like this:
+
+```bash
+projectfolder/
+   - app.py
+   - templates/
+   - static/
+```
+
+Now in app.py we'll add the following lines of code to create a server.
+
+```python
+from flask import Flask ,render_template
+
+app = Flask(__name__)
+
+@app.route("/")
+@app.route("/index")
+def index():
+	return render_template("index.html")
+
+if __name__ =='__main__':
+	app.run(debug=True)
+
+```
+
+The code block above creates a simple server with one endpoint (or route) that simple renders (or displays) a web page, which is supposed to be from a file called `index.html` as seen in the code. However, we haven't created that file yet. Remember, I mentioned that all HTML files go in the templates folder we created earlier. Hence, we'll go to that folder and add a new file called `index.html` so our flask server can see it. In `index.html` we can add a simple "hello world" h1 tag as follows:
```

Comment:
that simply*

You can try avoiding words like simply or basically

Reply:
Thanks I'll fix it now

Reply:
Done